CRM/Badge/Form/Layout.php

```diff
@@ -154,7 +154,7 @@ public function buildQuickForm() {
   public function setDefaultValues() {
     if (isset($this->_id)) {
       $defaults = array_merge($this->_values,
-        CRM_Badge_BAO_Layout::getDecodedData($this->_values['data']));
+        CRM_Badge_BAO_Layout::getDecodedData(CRM_Utils_Array::value('data', $this->_values, '[]')));
```

Comment:
This checks for data in $this->_values and passes an empty json_encoded array as the default value (getDecodedData does a jsonDecode